sycl/include/sycl/detail/ur.hpp

```diff
@@ -94,7 +94,7 @@ template <class To, class From> To cast(From value);
 template <class To, class From> inline To cast(From value) {
   // TODO: see if more sanity checks are possible.
   static_assert(sizeof(From) == sizeof(To), "assert: cast failed size check");
-  return reinterpret_cast<To>(value);
+  return (To)(value);
```

Comment:
`reinterpret_cast` will create a compilation problem where some native handles have a cast from unsigned long long to unsigned long (size_t) and I believe this could happen as backends have different sizes for its handles. So, this will make it as it was in `pi.hpp` temporarily until there is a proper fix for this part if possible.

The static cast check before this part didn't fail as `From` and `To` are pointers so would have the same size but the pointed type could be different.

Reply:
does this https://github.com/intel/llvm/pull/14830 fix these tests as well? It would be better not to reintroduce the C-style cast

Reply:
Yeah, good shout, that fixes it. Will close it and redirect it to the other PR